### ISSUE TITLE
Install/Config issue

### ISSUE THREAD

**Issue by stever145** [gormanb/homebridge-hivehome-control#5]
Unable to get the plugin working with my Hive config.  Have tried uninstalling/re-installing etc.  

The first error I get in the log is:
TypeError: this.hiveAccessories[deviceType] is not a constructor
    at HiveHomeControllerPlatform.discoverDevices (/usr/local/lib/node_modules/homebridge-hivehome-control/src/platform.ts:147:11)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)

But then, I get a series of errors as follows:

[11/4/2024, 7:19:14 PM] Registering platform 'homebridge-hivehome-control.HiveHomeController'
[11/4/2024, 7:19:14 PM] [homebridge-hivehome-control] Loaded homebridge-hivehome-control v1.0.3 child bridge successfully
[11/4/2024, 7:19:14 PM] Loaded 1 cached accessories from cachedAccessories.0EDD4662DCC4.
[11/4/2024, 7:19:14 PM] [homebridge-hivehome-control] Loading accessory from cache: Bedroom undefined
[11/4/2024, 7:19:14 PM] Homebridge v1.8.4 (HAP v0.12.2) (homebridge-hivehome-control) is running on port 34829.
[11/4/2024, 7:19:14 PM] 

NOTICE TO USERS AND PLUGIN DEVELOPERS
> Homebridge 2.0 is on the way and brings some breaking changes to existing plugins.
> Please visit the following link to learn more about the changes and how to prepare:
> https://github.com/homebridge/homebridge/wiki/Updating-To-Homebridge-v2.0

** Error in exception handler ** Error: ENOENT: no such file or directory, open '/usr/local/lib/node_modules/homebridge-hivehome-control/src/hivehome/hive-helpers.ts'
    at Object.readFileSync (node:fs:448:20)
    at processJSStacktrace (/usr/local/lib/node_modules/homebridge-hivehome-control/node_modules/pythonia/src/pythonia/errors.js:68:25)
    at Object.getErrorMessage (/usr/local/lib/node_modules/homebridge-hivehome-control/node_modules/pythonia/src/pythonia/errors.js:83:24)
    at new PythonException (/usr/local/lib/node_modules/homebridge-hivehome-control/node_modules/pythonia/src/pythonia/Bridge.js:21:25)
    at Bridge.call (/usr/local/lib/node_modules/homebridge-hivehome-control/node_modules/pythonia/src/pythonia/Bridge.js:251:37)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at startHiveSession (/usr/local/lib/node_modules/homebridge-hivehome-control/src/hivehome/hive-helpers.ts:38:17)
    at HiveHomeControllerPlatform.discoverDevices (/usr/local/lib/node_modules/homebridge-hivehome-control/src/platform.ts:111:25) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/usr/local/lib/node_modules/homebridge-hivehome-control/src/hivehome/hive-helpers.ts'
}
** JavaScript Stacktrace **
    at Bridge.call (/usr/local/lib/node_modules/homebridge-hivehome-control/node_modules/pythonia/src/pythonia/Bridge.js:251:37)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at startHiveSession (/usr/local/lib/node_modules/homebridge-hivehome-control/src/hivehome/hive-helpers.ts:38:17)
    at HiveHomeControllerPlatform.discoverDevices (/usr/local/lib/node_modules/homebridge-hivehome-control/src/platform.ts:111:25)
** Python Stacktrace **
Traceback (most recent call last):
  File "/usr/local/lib/node_modules/homebridge-hivehome-control/node_modules/pythonia/src/pythonia/Bridge.py", line 365, in onMessage
    return getattr(self, action)(r, ffid, key, args)
           ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/node_modules/homebridge-hivehome-control/node_modules/pythonia/src/pythonia/Bridge.py", line 338, in pcall
    self.call(r, ffid, key, pargs, kwargs or {})
    ~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/node_modules/homebridge-hivehome-control/node_modules/pythonia/src/pythonia/Bridge.py", line 198, in call
    v = v(*args, **kwargs)
  File "/usr/local/lib/node_modules/homebridge-hivehome-control/pylib/pyhiveapi/session.py", line 231, in login
    result = self.auth.login()
  File "/usr/local/lib/node_modules/homebridge-hivehome-control/pylib/pyhiveapi/api/hive_auth.py", line 359, in login
    if response["ChallengeName"] == self.PASSWORD_VERIFIER_CHALLENGE:
       ~~~~~~~~^^^^^^^^^^^^^^^^^
TypeError: 'NoneType' object is not subscriptable
node:internal/process/promises:391
    triggerUncaughtException(err, true /* fromPromise */);
    ^
[PythonException [Error]]
Node.js v20.18.0





**Plugin Config:**

config.json:

{
    "hiveUsername": "---------------",
    "hivePassword": "------------",
    "deviceGroupKey": "--------",
    "deviceKey": "eu-west-1_b7d9980c-95b2-4e49-87fd--------",
    "devicePassword": "----------------------------------------",
    "hotWaterBoostMins": 60,
    "heatingBoostMins": 60,
    "heatingBoostTemp": 21,
    "enableDebugLog": false,
    "platform": "HiveHomeController",
    "_bridge": {
        "username": "0E:DD:46:62:DC:C4",
        "port": 34829
    }
}

Plugin version: 1.0.3
Homebridge: 1.8.4
Node.js: 20.18.0
NPM: 10.8.2
OS: MacOS Sequoia

**Comment by Mercifull** [comment]
I have had a similar nightmare installing this plugin as well. I managed to resolve some of the issues by installing some other bits manually and reinstalling plugin but i too am hitting the brick wall at:
`** Error in exception handler ** Error: ENOENT: no such file or directory, open '/usr/local/lib/node_modules/homebridge-hivehome-control/src/hivehome/hive-helpers.ts'`

**Comment by Mercifull** [comment]
Have you double/triple/quadruple checked the device id or device password? It turns out that I was using a 1 instead of a lowercase l in the long device string and once I corrected it the plugin started working.

**Comment by stever145** [comment]
The deviceGroupkey, deviceKey and devicePassword are all copy/paste from the other window, so should all be OK.  Same for hivePassword - it's copied from my Mac's password app.

**Comment by stever145** [comment]
I do notice that from the logs, it says: Loading accessory from cache: Bedroom undefined

Bedroom is one of the Hive Thermostatic Radiator valves, so I wonder if the plugin doesn't recognise/support them (I'm not bothered about that - I just want it to be able to boost my water/heating from within Apple Home

**Comment by Mercifull** [comment]
Ahh ok. Prior to me realising this I also ran these two commands in terminal. Did they make a difference? Who knows. Might be worth a shot though. 

sudo apt-get update
sudo apt-get install -y libxslt1-dev

**Comment by Mercifull** [comment]
The description of this app specifically says it’s designed only for those boost functions so I’m surprised it’s trying to load the thermostatic valve you might be right there that it’s causing the issue.

**Comment by tommylux** [comment]
Getting same issue.. Can anyone advise if:

- Device Group key should contain leading hypen?
- Device key includes eu-west-1 prefix?
- Device password has two trailing equal sign?

Thanks

**Comment by Mercifull** [comment]
My device group key DOES contain a leading hyphen
My device key DOES begin eu-west-1_
My device password DOES end ==

**Comment by tommylux** [comment]
Thanks for confirming, does this plugin work for you? I can't control anything in Hive.. Thanks

**Comment by Mercifull** [comment]
It works… or at least it seems to work fine. However I am getting loads of errors in the log about various files not found and things like that. I only have one device on my hive and that’s the central heating.

**Comment by gormanb** [comment]
@stever145: it looks like this is indeed a problem with the plugin not recognizing TRVs. Essentially, the library that the plugin uses to communicate with the Hive server conflates TRVs and Hive thermostats into a single "climate" category, but the plugin only expects to see the latter device type.

From what I can tell the TRV controller also has a Boost function, so I've released a new beta `1.0.4-beta.0` which should add support for TRVs. Since I don't have any TRVs I can't test this myself, so I'd appreciate it if you could:
- Install the new beta and confirm that you no longer see the `TypeError: this.hiveAccessories[deviceType] is not a constructor` error
- Confirm that your TRV appears in Homekit as a separate device (i.e. switch)
- Let me know if everything works as expected when you enable and disable this switch.

**Comment by stever145** [comment]
Hi - no longer seeing the error, and I am able to boost my heating via the Apple Home app, along with altering the TRV etc.  The status appears to be correctly flowing back from Hive to Apple Home too.  Thanks for looking into this.

**Comment by gormanb** [comment]
Thanks for confirming @stever145! I've just published `1.0.4` as a full release version, so please feel free to install that instead of the beta.
